tests/gpu/displaylist/alignment.cpp

```diff
@@ -0,0 +1,253 @@
+	if (listid < 0) {
+		return listid;
+	}
+	sceGeListUpdateStallAddr(listid, listend);
+	if (head) {
+		sceGeContinue();
+	}
+	sceKernelDelayThread(5000);
+	sceGeBreak(1, NULL);
+
+	return sceGeGetCmd(GE_CMD_AMBIENTCOLOR);
+}
+
+u32 runListAndReset(void *list, void *listend, bool head = false) {
+	u32 result = runList(list, listend, head);
+	runList(dlistReset, dlistReset + 100);
+	return result;
+}
+
+u32 runJumpList(int offset) {
+	dlist2[3] = (GE_CMD_JUMP << 24) | (0x000018 + offset);
+	sceKernelDcacheWritebackInvalidateRange(dlist2, sizeof(dlist2));
+
+	return runList(dlist2, dlist2 + 100);
+}
+
+u32 runBJumpList(int offset) {
+	dlist3[4] = (GE_CMD_BJUMP << 24) | (0x000018 + offset);
+	sceKernelDcacheWritebackInvalidateRange(dlist3, sizeof(dlist3));
+
+	return runList(dlist3, dlist3 + 100);
+}
+
+u32 runCallList(int offset) {
+	dlist4[3] = (GE_CMD_CALL << 24) | (0x000014 + offset);
+	sceKernelDcacheWritebackInvalidateRange(dlist4, sizeof(dlist4));
+
+	return runList(dlist4, dlist4 + 100);
+}
+
+u32 runSignalJumpList(int offset) {
+	u32 target = (intptr_t)dlist5 + 0x20 + offset;
+	dlist5[3] = (GE_CMD_SIGNAL << 24) | (0x10 << 16) | (target >> 16);
+	dlist5[4] = (GE_CMD_END << 24) | (target & 0xFFFF);
+	sceKernelDcacheWritebackInvalidateRange(dlist5, sizeof(dlist5));
+
+	return runList(dlist5, dlist5 + 100);
+}
+
+u32 runSignalCallList(int offset) {
+	u32 target = (intptr_t)dlist6 + 0x1C + offset;
+	dlist6[3] = (GE_CMD_SIGNAL << 24) | (0x10 << 16) | (target >> 16);
+	dlist6[4] = (GE_CMD_END << 24) | (target & 0xFFFF);
+	sceKernelDcacheWritebackInvalidateRange(dlist6, sizeof(dlist6));
+
+	return runList(dlist6, dlist6 + 100);
+}
+
+extern "C" int main(int argc, char *argv[]) {
+	checkpointNext("Enqueue");
+	checkpoint("  Aligned: %x", runListAndReset(dlist1, dlist1 + 100));
+	checkpoint("  Offset + 1: %x", runListAndReset((u8 *)dlist1 + 1, dlist1 + 100));
+	checkpoint("  Offset + 2: %x", runListAndReset((u8 *)dlist1 + 2, dlist1 + 100));
+	checkpoint("  Offset + 3: %x", runListAndReset((u8 *)dlist1 + 3, dlist1 + 100));
+	checkpoint("  Offset + 4: %x", runListAndReset((u8 *)dlist1 + 4, dlist1 + 100));
+	checkpoint("  Offset + 8: %x", runListAndReset((u8 *)dlist1 + 8, dlist1 + 100));
+
+	checkpointNext("EnqueueHead");
+	checkpoint("  Aligned: %x", runListAndReset(dlist1, dlist1 + 100, true));
+	checkpoint("  Offset + 1: %x", runListAndReset((u8 *)dlist1 + 1, dlist1 + 100, true));
+	checkpoint("  Offset + 2: %x", runListAndReset((u8 *)dlist1 + 2, dlist1 + 100, true));
+	checkpoint("  Offset + 3: %x", runListAndReset((u8 *)dlist1 + 3, dlist1 + 100, true));
+	checkpoint("  Offset + 4: %x", runListAndReset((u8 *)dlist1 + 4, dlist1 + 100, true));
+	checkpoint("  Offset + 8: %x", runListAndReset((u8 *)dlist1 + 8, dlist1 + 100, true));
+
+	checkpointNext("Jump");
+	checkpoint("  Aligned: %x", runJumpList(0));
+	checkpoint("  Offset + 1: %x", runJumpList(1));
+	checkpoint("  Offset + 1: %x", runJumpList(2));
```

Comment:
Not really a problem but there's a typo in the string, should be 2 (and next one 3), of course.